pkg/chartverifier/checks/charttesting.go

```diff
@@ -160,16 +169,24 @@ func generateInstallConfig(
 	chrt *chart.Chart,
 	helm tool.Helm,
 	kubectl tool.Kubectl,
+	configRelease string,
 ) (namespace, release, releaseSelector string, cleanup func()) {
+	release = configRelease
 	if cfg.Namespace != "" {
 		namespace = cfg.Namespace
-		release, _ = chrt.CreateInstallParams(cfg.BuildId)
+		if len(release) == 0 {
+			release, _ = chrt.CreateInstallParams(cfg.BuildId)
+		}
 		releaseSelector = fmt.Sprintf("%s=%s", cfg.ReleaseLabel, release)
 		cleanup = func() {
 			helm.DeleteRelease(namespace, release)
 		}
 	} else {
-		release, namespace = chrt.CreateInstallParams(cfg.BuildId)
+		if len(release) == 0 {
+			release, namespace = chrt.CreateInstallParams(cfg.BuildId)
+		} else {
+			_, namespace = chrt.CreateInstallParams(cfg.BuildId)
+		}
```

Comment:
Seems the if blocks are duplicated, can you put the `len(release)` check at the begining of this function instead of repeating in each condition?

Reply:
Blocks are subtly different in what they get back from chrt.CreateInstallParams(cfg.BuildId). There are 4 things to cover:
1. Namespace and Release both set
2. Namespace set and Release not set
3. Namespace not set and Release set
4. Neither Namespace nor Release set

chose nested ifs insteand of a if, else if, else if, else

Reply:
What I propose was:
```
		if len(release) == 0 {
			release, namespace = chrt.CreateInstallParams(cfg.BuildId)
		} else {
			_, namespace = chrt.CreateInstallParams(cfg.BuildId)
		}
        if cfg.Namespace != "" {
            ...
        } else {
            ...
        }
```
`chrt.CreateInstallParams(cfg.BuildId)` was called at the begining of each block so not sure why they cannot be hoist up to the beginning to the function. But I think this is just personally preference, and definitely not a blocker.

Reply:
my con with your code is that we may call chrt.CreateInstallParams(cfg.BuildId) when we don't need to because namespace was set in the cfg. Anyway we agree that there are a ton of ways to write this and all have pros ad cons. I kind of went with the minimum change on the current logic.

Reply:
also the existing code is well tested and has value based on how many times it has been run and worked. On this basis I try to not change more than I need to.

Reply:
agreed, resolving